core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -251,12 +255,22 @@ object ConsoleProducer {
     val options = parser.parse(args : _*)
     if(args.length == 0)
       CommandLineUtils.printUsageAndDie(parser, "Read data from standard input and publish it to Kafka.")
-    CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
 
     val useOldProducer = options.has(useOldProducerOpt)
-    val topic = options.valueOf(topicOpt)
+    val bootstrapServer = options.valueOf(bootstrapServerOpt)
```

Comment:
Ohh sorry, yeah you are right. I have one question, why can the user use the old-producer with the bootstrap-server? The broker-list and the bootstrap-server uses different properties "metadata.broker.list" vs "bootstrap.servers".